scripts/automation/Radius/Functions/Public/Start-GenerateUserCerts.ps1

```diff
@@ -0,0 +1,223 @@
+                        }
+                        $false {
+                            switch ($PSCmdlet.ParameterSetName) {
+                                'gui' {
+                                    $result = Invoke-UserCertProcess -radiusMember $userObject -certType $CertType -Prompt
+                                }
+                                'cli' {
+                                    $result = Invoke-UserCertProcess -radiusMember $userObject -certType $CertType
+                                }
+                            }
+                        }
+                    }
+                    Show-RadiusProgress -completedItems $userObject.count  -totalItems $userObject.count -ActionText "Generating Radius Certificates" -previousOperationResult $result
+                }
+                switch ($PSCmdlet.ParameterSetName) {
+                    'gui' {
+                        Show-StatusMessage -Message "Finished Generating Certificates"
+                    }
+                    'cli' {
+                        return
+                    }
+                }
+            }
+
+            '3' {
+                # re-generate new certificates for ALL users; will force rewrite all certs
+                switch ($PSCmdlet.ParameterSetName) {
+                    'gui' {
+                        $overwriteExistingCerts = Get-ResponsePrompt -message "Are you confident you want to replace all locally generated user certificates?"
+                        switch ($overwriteExistingCerts) {
+                            $true {
+                                continue
+                            }
+                            $false {
+                                return
+                            }
+                        }
+                    }
+                }
+                # Get each RadiusMember User:
+                for ($i = 0; $i -lt $JCRRadiusMembers.count; $i++) {
+                    $result = Invoke-UserCertProcess -radiusMember $JCRRadiusMembers[$i] -certType $CertType -forceReplaceCert
+                    Show-RadiusProgress -completedItems ($i + 1) -totalItems $JCRRadiusMembers.count -ActionText "Generating Radius Certificates" -previousOperationResult $result
+                }
+                switch ($PSCmdlet.ParameterSetName) {
+                    'gui' {
+                        Show-StatusMessage -Message "Finished Generating Certificates"
+                    }
+                    'cli' {
+                        return
+                    }
+                }
+            }
+            '4' {
+                # TODO: if there are no certs set to expire in 'x' days inform when pressing this option
+                # recalculate expiring certs:
+                $userCertInfo = Get-CertInfo -UserCerts
+                $global:cutoffDate = (Get-Date).AddDays(15).Date
+                $Global:expiringCerts = Get-ExpiringCertInfo -certInfo $userCertInfo -cutoffDate $cutoffDate
```

Comment:
This global var is being set and used by another function.

Reply:
CutOffDate has been moved and set just once. 

The # of days a is now configurable within `config.ps1` and the actual date is calculated when running the module (within the .`psm1` file) 

I believe this is resolved in daeca953583f2fbd5960ff586716c01d7e8e5794